### DISCUSSION TITLE
How to pass flags to scripted `.stroke`?

### DISCUSSION THREAD

**Discussion by paulbernhard** [fontforge/fontforge#4921] [Q&A]
Hello,

i am working on a script to expand strokes of entire fonts, using a python script and (among others) the `font.stroke()` function which is further explained under [`glyph.stroke()` in the scripting docs](https://fontforge.org/docs/scripting/python/fontforge.html#fontforge.glyph.stroke). While i have no problems using `stroke()` just with a `nib` and `width` argument, i do not manage to pass further flag arguments to the function without raising errors. This might not be a technical issue but the usage isn't clear to me from the docs.

The following works without a problem:

```
font = fontforge.open(sys.argv[1])
font.selection.all()
font.stroke("circular", int(weight))
```

The following throws a `ValueError: Unknown linecap type "nib"` while the documentation reads *CAP is optional. It can be specified either positionally or with cap=string. It must be one of the strings “nib” (the default), “butt”, “round”, and “bevel”.*. In this case passing `"butt"` as third argument would work.

```
font.stroke("circular", int(weight), "nib")
```

When trying to pass the `KEYWORD` arguments, i am getting a `TypeError: stroke() takes no keyword arguments`.

```
font.stroke("circular", int(weight), "nib", "nib", removeinternal=True)
```

… or `ValueError: Unknown stroke flag "removeinternal=true"` when trying the following.

```
font.stroke("circular", int(weight), "butt", "round", "removeinternal=true")
```

Could anybody explain the usage of those arguments, please?

Fontforge version: fontforge 20200314
OS: Ubuntu 20.04.3 LTS

**Comment by iterumllc** [comment]
This looks suspiciously to me like you've somehow got a pre-20200314 fontforge implementation in your PYTHONPATH. What does `python -c "import fontforge; print(fontforge.__version__)"` print?

**Comment by paulbernhard** [comment]
I was running above's command `import fontforge; print(fontforge.__version__)` to get the fontforge module version, resulting in:

``` 
20190801 git:
```

Indeed, the version of my python module seems to be pre 2020. It was installed via the `python3-fontforge` package while i am using fontforge via the recommended AppImage (currently `FontForge-2020-03-14-67687b0-x86_64.AppImage`). Do you know how i could get the more up-to-date `.AppImage` version to load in my python modules? Or would it be better to switch to using a manager like pip?

(I am not very firm with python and/or its package manager.)

**Reply by iterumllc** [reply]
Trying to load modules embedded within an appimage (assuming they're there) is tricky. If your script doesn't have other exotic imports you could try running it through fontforge instead of the python application, as in `fontforge -lang=py -script [scriptname]`.
